dissect/target/plugins/os/windows/catroot.py

```diff
@@ -60,67 +114,135 @@ def catroot(self):
+        for ese_file in catroot_info.base_path.rglob(catroot_info.pattern):
+            with ese_file.open("rb") as fh:
+                ese_db = EseDB(fh)
+
+                tables = [table.name for table in ese_db.tables()]
+                for hash_type, table_name in [("sha256", "HashCatNameTableSHA256"), ("sha1", "HashCatNameTableSHA1")]:
+                    if table_name not in tables:
                         continue
 
-                    for offset in findall(buf, needle):
-                        try:
-                            digestid = algos.DigestAlgorithmId.load(buf[offset:])
-                            # 4 bytes before the digest type, a sequence starts
-                            objseq = core.Sequence.load(buf[offset - 4 :])
-                            # The second entry in the sequence is the digest string
-                            hexdigest = objseq[1].native.hex()
-
-                            # Later versions of windows also have a file hint
-                            # Try to find it
-                            digestlen = len(digestid.contents)
-                            hintoffset = buf.find(HINT_NEEDLE, offset + digestlen, offset + digestlen + 64)
-
-                            filehint = None
-                            if hintoffset != -1:
-                                try:
-                                    file_buf = buf[hintoffset + len(HINT_NEEDLE) + 6 :]
-                                    # There's an INTEGER after the Hint BMPString of size 6
-                                    filehint = core.OctetString.load(file_buf).native.decode("utf-16-le")
-                                except Exception:
-                                    pass
-
-                            fdigest = digest()
-                            if len(hexdigest) == 32:
-                                fdigest.md5 = hexdigest
-                            elif len(hexdigest) == 40:
-                                fdigest.sha1 = hexdigest
-                            elif len(hexdigest) == 64:
-                                fdigest.sha256 = hexdigest
-
-                            yield CatrootRecord(
-                                digest=fdigest,
-                                hint=self.target.fs.path(filehint) if filehint else None,
-                                source=f,
-                                _target=self.target,
-                            )
-                        except Exception:
-                            continue
+                    for record in ese_db.table(table_name).records():
+                        file_digest = digest()
+                        setattr(file_digest, hash_type, record.get("HashCatNameTable_HashCol").hex())
+                        raw_hint = record.get("HashCatNameTable_CatNameCol").decode("utf-8").rstrip("|")
+
+                        yield CatrootRecord(
+                            digest=file_digest,
+                            hints=raw_hint.split("|"),
```

Comment:
It doesn't look like these are the actual hints? More the source .cat files. Maybe the DB doesn't have the hints?